examples/gno.land/r/demo/keystore/keystore.gno

```diff
@@ -0,0 +1,189 @@
+package keystore
+
+import (
+	"std"
+	"strings"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/demo/ufmt"
+)
+
+var data avl.Tree
+
+var (
+	BaseURL               = "/r/demo/keystore"
+	StatusOK              = "ok"
+	StatusNoUser          = "user not found"
+	StatusNotFound        = "key not found"
+	StatusNoWriteAccess   = "no write access"
+	StatusCouldNotExecute = "could not execute"
+)
+
+func init() {
+	data = avl.Tree{} // user -> avl.Tree
+}
+
+// KeyStore stores the owner-specific avl.Tree
+type KeyStore struct {
+	Owner std.Address
+	Data  avl.Tree
+}
+
+// Set will set a value to a key
+// requires write-access (original caller must be caller)
+func Set(k, v string) string {
+	origOwner := std.GetOrigCaller()
+	return set(origOwner.String(), k, v)
+}
+
+// set (private) will set a key to value
+// requires write-access (original caller must be caller)
+func set(owner, k, v string) string {
+	origOwner := std.GetOrigCaller()
+	if origOwner.String() != owner {
+		return StatusNoWriteAccess
+	}
+	keystoreInterface, exists := data.Get(owner)
```

Comment:
What do you think about extracting this out into a specific helper method?

Reply:
I'm of the mind to leave it - it only occurs twice in the code and moving out into a function may only reduce the code by a few lines.